### PR TITLE
Clean up spinnaker-jenkins-to-vmss template

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
The first commit is just formatting the files (based on default formatting from VS Code - I _think_ the most popular tool for these templates), so it makes it look like more changes than it is. I recommend looking at the 3 commits separately. This is a first step in making the template ready for our CI system (basically the simpler the template, the easier it is to integrate)

### CODE REVIEW

**Comment by EricJizbaMSFT** [comment]
@dcaro @rguthriemsft PTAL

File: spinnaker-jenkins-to-vmss/azuredeploy.json [hunk 1]
```diff
@@ -1,533 +1,516 @@
...
-    "spinnakerNICName" : "spinnakerNic",
-    "spinnakerPrivateIP" : "10.0.0.4",
+    "spinnakerNSGName": "spinnakerNSG",
+    "spinnakerNICName": "spinnakerNic",
     "spinnakerPublicIPAddressName": "spinnakerPublicIP",
     "spinnakerVMName": "spinnakerVm",
     "spinnakerOSDiskName": "spinnakerOSDisk",
-    "keysPermissions": [ "all" ],
-    "secretsPermissions": [ "get" ]
+    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/spinnaker-jenkins-to-vmss/",
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.json]: See BP guide for how to handle artifacts

File: spinnaker-jenkins-to-vmss/azuredeploy.json [hunk 2]
```diff
@@ -1,533 +1,516 @@
...
+      "metadata": {
+        "description": "Name of your Oracle Account used to download the JDK."
+      }
     },
+    "JenkinsUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of your Jenkins User account."
+      },
+      "defaultValue": "jenkins"
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.json]: remove default value to reduce surface area

File: spinnaker-jenkins-to-vmss/azuredeploy.json [hunk 3]
```diff
@@ -1,533 +1,516 @@
...
-    "vnetAddressSpace" : "10.0.0.0/20",
     "subnet0Name": "[concat(variables('scenarioPrefix'),'Subnet0')]",
-    "subnet0AddressPrefix": "10.0.0.0/24",
     "subnet1Name": "[concat(variables('scenarioPrefix'),'Subnet1')]",
-    "subnet1AddressPrefix": "10.0.1.0/24",
     "subnet2Name": "[concat(variables('scenarioPrefix'),'Subnet2')]",
-    "subnet2AddressPrefix": "10.0.2.0/24",
-    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('vnetName'))]",
-    "subnet0Ref": "[concat(variables('vnetID'),'/subnets/', variables('subnet0Name'))]",
+    "subnet0Ref": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('vnetName')),'/subnets/', variables('subnet0Name'))]",
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.json]: just use resourceId, no need for concat

File: spinnaker-jenkins-to-vmss/azuredeploy.json [hunk 4]
```diff
@@ -1,533 +1,516 @@
...
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('jenkinsNICName'))]"
             }
+          ]
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.json]: consider using a child resource

File: spinnaker-jenkins-to-vmss/azuredeploy.json [hunk 5]
```diff
@@ -1,533 +1,516 @@
...
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": "true",
+            "storageUri": "[concat(reference(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2016-01-01').primaryEndpoints.blob)]"
+          }
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.json]: consider using a child resource

File: spinnaker-jenkins-to-vmss/azuredeploy.parameters.json [hunk 6]
```diff
@@ -1,9 +1,9 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
   "contentVersion": "1.0.0.0",
   "parameters": {
     "vmUserName": {
-      "value": "GEN-USERNAME"
+      "value": "azureuser"
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.parameters.json]: use GEN-UNIQUE

File: spinnaker-jenkins-to-vmss/azuredeploy.parameters.json [hunk 7]
```diff
@@ -20,20 +20,17 @@
...
-      "value": "GEN-UNIQUE"
-    },
-    "OracleUserName" : {
+    "OracleUserName": {
       "value": "your-oracle-account-identifier"
     },
-    "JenkinsUserName" : {
-      "value": "GEN-USERNAME"
+    "JenkinsUserName": {
+      "value": "jenkins"
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.parameters.json]: GEN-UNIQUE

File: spinnaker-jenkins-to-vmss/azuredeploy.parameters.json [hunk 8]
```diff
@@ -20,20 +20,17 @@
...
+    },
+    "JenkinsPassword": {
```
Comment (bmoore-msft) [inline spinnaker-jenkins-to-vmss/azuredeploy.parameters.json]: GEN-PASSWORD

**Pull request closed without merging**